File: template/template.rb
require 'uri'

url = "{{ .URL }}"

p uri = URI.parse(url)

p 'scheme: ' + uri.scheme    # => "http"
p 'userinfo: ' + (uri.userinfo.nil? ? 'nil' : uri.userinfo)
p 'host: ' + (uri.host.nil? ? 'nil' : uri.host)       # => "www.ruby-lang.org"
p 'port: ' + (uri.port.nil? ? 'nil' : uri.port.to_s)      # => 80
p 'registry: ' + (uri.registry.nil? ? 'nil' : uri.registry)
p 'path: ' + (uri.path.nil? ? 'nil' : uri.path)
p 'opaque: ' + (uri.opaque.nil? ? 'nil' : uri.opaque)
p 'query: ' + (uri.query.nil? ? 'nil' : uri.query)
p 'fragment: ' + (uri.fragment.nil? ? 'nil' : uri.fragment)
#p 'arg_check: ' + (uri.arg_check.nil? ? 'nil' : uri.arg_check)

File: template/template.pl
#!/usr/bin/perl
use URI;

$url = '{{ .URL }}';
$u = URI->new($url);
print 'scheme = ' . $u->scheme . "\n";
print 'userinfo = ' . $u->userinfo . "\n";
print 'host = ' . $u->host . "\n";
print 'port = ' . $u->port . "\n";
print 'opaque = ' . $u->opaque . "\n";
print 'path = ' . $u->path . "\n";
print 'query = ' . $u->query . "\n";
print 'fragment = ' . $u->fragment . "\n";
print 'canonical = ' . $u->canonical . "\n";
